### PR TITLE
Conditionally render legacy reject dialog

### LINKED ISSUE #1206: Support pre-demo reject UI
### Description
Closes #1096. Adds the ability for the backoffice to use the pre-demo reject flow via workflow definition config.

### PR DESCRIPTION
**Pull request by Omri-Levy**
### Description
Now using the workflow definition's version to determine which reject UI to use, this PR also fixes the bugs related to using the reject dialog.

Updated reject:
image

Updated reject dialog:
image

### CODE REVIEW

File: apps/backoffice-v2/src/common/utils/handle-zod-error/handle-zod-error.ts [hunk 1]
```diff
@@ -2,7 +2,7 @@ import { isZodError } from '../is-zod-error/is-zod-error';
 
 export const handleZodError = <TData>(error: unknown, data: TData) => {
   if (!error || isZodError(error)) {
-    return data;
+    return data ?? null;
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/common/utils/handle-zod-error/handle-zod-error.ts:5]: Fixes long-standing issue.  `react-query` is expecting `null` and warns against `undefined`.

File: apps/backoffice-v2/src/domains/workflows/fetchers.ts [hunk 2]
```diff
@@ -178,7 +179,9 @@ export const fetchWorkflowDecision = async ({
     endpoint: `workflows/${workflowId}/decision/${documentId}`,
     method: Method.PATCH,
     body,
-    schema: WorkflowByIdSchema,
+    schema: WorkflowByIdSchema.pick({
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/domains/workflows/fetchers.ts:183]: Currently it's a substantial amount of effort to format the response workflow the same way as in `getWorkflowByIdWithRelations`, the way this fetcher is used `context` is the only prop needed, and doing so resolves errors on revision/approve for a task.

File: apps/backoffice-v2/src/pages/Entity/components/CallToAction/CallToAction.tsx [hunk 3]
```diff
@@ -45,112 +47,224 @@ export const CallToAction: FunctionComponent<ICallToActionProps> = ({ value, dat
...
-                    The case’s status will then change to “Revisions” until the customer will
-                    provide the needed documents and fixes.
-                  </span>
-                </>
-              )}
-            </p>
-          </DialogDescription>
-        </DialogHeader>
-        {!noReasons && (
+  if (value === 'Reject') {
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/CallToAction/CallToAction.tsx:50]: Pasted the dialog from the `dev` branch and updated the buttons to fit the new design direction.

File: apps/backoffice-v2/src/pages/Entity/components/CallToAction/hooks/useCallToActionLogic/useCallToActionLogic.tsx [hunk 4]
```diff
@@ -110,6 +110,10 @@ export const useCallToActionLogic = () => {
     workflow?.workflowDefinition?.config?.workflowLevelResolution ??
     workflow?.context?.entity?.type === 'business';
 
+  useEffect(() => {
```
Comment (Omri-Levy) [inline apps/backoffice-v2/src/pages/Entity/components/CallToAction/hooks/useCallToActionLogic/useCallToActionLogic.tsx:113]: Fixes long-standing bug of an empty dropdown when changing the reject action while still sending a wrong rejection reason.

File: services/workflows-service/src/workflow/workflow.service.ts [hunk 5]
```diff
@@ -903,6 +908,10 @@ export class WorkflowService {
         documents: context?.documents?.map(
           (document: DefaultContextSchema['documents'][number]) => ({
             ...document,
+            decision: {
+              ...document?.decision,
+              status: document?.decision?.status === null ? undefined : document?.decision?.status,
```
Comment (Omri-Levy) [inline services/workflows-service/src/workflow/workflow.service.ts:917]: The ability to cancel a request for re-upload introduced `status: null`. `null` is needed due to the nature of the `merge` method we use but `null` also breaks a number of places (validation).

**Review by MatanYadaev — APPROVED** [review]

**Pull request merged**